server/apiserver/argoserver.go

```diff
@@ -307,8 +308,8 @@ func (as *argoServer) newHTTPServer(ctx context.Context, port int, artifactServe
 	mux.HandleFunc("/input-artifacts/", artifactServer.GetInputArtifact)
 	mux.HandleFunc("/artifacts-by-uid/", artifactServer.GetOutputArtifactByUID)
 	mux.HandleFunc("/input-artifacts-by-uid/", artifactServer.GetInputArtifactByUID)
-	mux.HandleFunc("/oauth2/redirect", as.oAuth2Service.HandleRedirect)
-	mux.HandleFunc("/oauth2/callback", as.oAuth2Service.HandleCallback)
+	mux.Handle("/oauth2/redirect", handlers.ProxyHeaders(http.HandlerFunc(as.oAuth2Service.HandleRedirect)))
```

Comment:
Using the ProxyHeaders handler `r.URL.Scheme` will be populated if any of the x-proto headers are set by the proxy.